### PR TITLE
Add the `pipeline.yml` file for Buildkite

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
cc: @ChrisRackauckas

**Comment by maleadt** [comment]
FWIW, It's possible to do `1.6-nightly` instead of nightly. Also, adding a `skip ci ` or `skip tests` rule is often convenient (see the JuliaGPU/buildkite README).

**Comment by DilumAluthge** [comment]
Done and done.

**Comment by DilumAluthge** [comment]
For now, I'll comment out the regular nightly and only run the 1.6-nightly.

**Comment by maleadt** [comment]
You want `if: build.message !~ /\[skip tests\]/`, `skip ci` is handled by buildkite itself. Also, you didn't add it to the nightly target.

**Comment by maleadt** [comment]
And `1.6-nightly requires v0.6 of the julia plugin.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+env:
+  JULIA_PKG_SERVER: "" # it often struggles with our large artifacts
+  # SECRET_CODECOV_TOKEN: "..."
```
Comment (maleadt) [inline .buildkite/pipeline.yml:3]: FYI, The secret variables can be huge, once encrypted. That's why I had put it at the bottom. (see e.g. CUDA.jl)

Comment (DilumAluthge) [reply .buildkite/pipeline.yml:3]: Ah fair point.

Comment (DilumAluthge) [reply .buildkite/pipeline.yml]: Done.

**Pull request merged**